metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/loader/api/SraAnalysisDocumentAndFileTypeLoaderTest.java

```diff
@@ -0,0 +1,119 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.loader.api;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import uk.ac.ebi.ena.sra.xml.ANALYSISDocument;
+import uk.ac.ebi.ena.sra.xml.AnalysisFileType;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SraAnalysisDocumentAndFileTypeLoaderTest {
+
+    private static final String ANALYSIS_ACCESSION = "ERZ496533";
+    private static final String ANALYSIS_INVALID_ACCESSION = "ERZ4965336";
+    private static final String ANALYSIS_DOCUMENT_XML = "AnalysisDocument.xml";
+    private static final String ANALYSIS_DOCUMENT_NOT_FOUND = "AnalysisDocumentNotFound.xml";
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    private SraAnalysisDocumentLoader sraAnalysisDocumentLoader;
+
+    private SraAnalysisFileTypeLoader sraAnalysisFileTypeLoader;
+
+    @Before
+    public void restTemplateSetup() throws Exception {
+        when(restTemplate.exchange(SraAnalysisDocumentLoader.getAnalysisXmlUrl(),
+                HttpMethod.GET, null, String.class, ANALYSIS_ACCESSION)).
+                thenReturn(new ResponseEntity(new String(Files.readAllBytes(Paths.get(getClass().getClassLoader()
+                        .getResource(ANALYSIS_DOCUMENT_XML).toURI()))), HttpStatus.OK));
+        when(restTemplate.exchange(SraAnalysisDocumentLoader.getAnalysisXmlUrl(),
+                HttpMethod.GET, null, String.class, ANALYSIS_INVALID_ACCESSION)).
+                thenReturn(new ResponseEntity(new String(Files.readAllBytes(Paths.get(getClass().getClassLoader()
+                        .getResource(ANALYSIS_DOCUMENT_NOT_FOUND).toURI()))), HttpStatus.OK));
+        sraAnalysisDocumentLoader = new SraAnalysisDocumentLoader(restTemplate);
+        sraAnalysisFileTypeLoader = new SraAnalysisFileTypeLoader(sraAnalysisDocumentLoader);
+    }
+
+    @Test
+    public void testAnalysisLoad() {
+        List<ANALYSISDocument> listOfAnalysisType =
+                sraAnalysisDocumentLoader.getSraAnalysisDocuments(Arrays.asList(ANALYSIS_ACCESSION));
+        assertEquals(1, listOfAnalysisType.size());
+
+        ANALYSISDocument analysisDocument = listOfAnalysisType.get(0);
+        String studyRefAccession = "ERP107353";
```

Comment:
final strings are defined for others, even this can be moved there to keep consistency.